packages/validators/src/schemas/componentDefinition.ts

```diff
@@ -22,14 +30,120 @@ export const ComponentDefinitionSchema = z.object({
   id: DefinitionPropertyKeySchema,
   variables: z.record(
```

Comment:
I realised last week that `ComponentDefinitionSchema` and `ComponentSettingsSchema` have the same variable definitions but we validate them separately. As a result, they are not correctly updated and this one is missing the correct `validations` and `defaultValue` properties. With your PR, we should update the comment in line 35 as you're adding `defaultValue`.
However, it would be great if we could somehow merge parts of this with `ComponentSettingsSchema` which defines all the same properties - the only difference is that componentSettings allows a bound/ unbound content pointer instead of plain strings for content variables.

Reply:
Are these the same variables? Seems `ComponentDefinitionSchema` is how its defined in the users code when they call `defineComponents`, and `ComponentSettingsSchema` is how these values are stored in the entry schema. So, you wouldn't want users to be able to specify the bound/unbound pointers in their `defineComponents` call.

Reply:
You're right that we don't want bound/ unbound pointers in the registration call. Though, besides that they are exactly the same as we save the registered definition as is in the component Settings and just overwrite the `defaultValue`. For that reason, it would be somehow good to also combine both schema as currently they are not correctly defined for that reason that we have two places trying to define the same (validations missing)

Reply:
Thanks, looks very promising :)